torch/_inductor/codegen/triton.py

```diff
@@ -2255,16 +2256,38 @@ def can_fuse(self, node1, node2):
         _, (numel2, rnumel2) = node2.group
 
         if node1.is_reduction() and node2.is_reduction():
-            return numel1 == numel2 and rnumel1 == rnumel2
+            reduction_can_fuse = numel1 == numel2 and rnumel1 == rnumel2
+            if not reduction_can_fuse:
+                fusion_log.debug(
```

Comment:
Does this affect compile time perf at all?

If so, wrap this in a check to see if the fusion_log is enabled like
fusion_log.isEnabledFor(logging.DEBUG)

Reply:
It's just a function call overhead:

https://github.com/python/cpython/blob/dbbe0eebf0a53fe5f9f9467b4bb85ec16989c0f0/Lib/logging/__init__.py#L1526

No string formatting overhead.

String literals are interned into the bytecode (co_const) during bytecode transpilation.

I don't think we should expect any noticeable perf diff.

Reply:
The linting practice recommended for logging is to use format string literals with formatting additional arguments, which is good enough for perf for the above reasons.